src/main/groovy/actions/TransferActions.groovy

```diff
@@ -22,17 +23,20 @@ import actions.LabwareActions
 class TransferActions {
 
     def static stamp(Labware sourceLabware, Labware destinationLabware, 
-        MaterialType materialType, List<String> copyMetadata = []) {
+        MaterialType materialType, List<String> copyMetadata = [],
+        List<Metadatum> newMetadataToAdd = []) {
```

Comment:
This will only allow the same metadata to be added to each of the materials, the point is that each material will get a different value.